src/module.c

```diff
@@ -1054,16 +1114,123 @@ int RMCreateSearchCommand(RedisModuleCtx *ctx, const char *name,
+}
+
+#define CONFIG_SUBCOMMANDS(command_, func_, readonly_)                                                                                                                           \
+  SubCommand subcommands[] = {                                                                                                                                      \
+    {.name = "GET", .fullName = command_ "|GET", .flags = readonly_, .handler = func_, .setCommandInfo = SetFtConfigGetInfo, .position = {0, 0, 0}},    \
+    {.name = "SET", .fullName = command_ "|SET", .flags = "write", .handler = func_, .setCommandInfo = SetFtConfigSetInfo, .position = {0, 0, 0}},       \
+    {.name = "HELP", .fullName = command_ "|HELP", .flags = readonly_, .handler = func_, .setCommandInfo = SetFtConfigHelpInfo, .position = {0, 0, 0}}, \
+  }
+
+#define RS_READ_ONLY_FLAGS_DEFAULT IsEnterprise() ? "readonly " PROXY_FILTERED : "readonly"
+static int RegisterConfigSubCommands(RedisModuleCtx* ctx, RedisModuleCommand *configCommand) {
+  CONFIG_SUBCOMMANDS(RS_CONFIG, ConfigCommand, RS_READ_ONLY_FLAGS_DEFAULT);
+  return CreateSubCommands(ctx, configCommand, subcommands, sizeof(subcommands) / sizeof(SubCommand));
+}
+
+static int RegisterCoordConfigSubCommands(RedisModuleCtx* ctx, RedisModuleCommand *configCommand) {
+  RedisModuleCmdFunc func = SafeCmd(ConfigCommand);
+  CONFIG_SUBCOMMANDS("FT.CONFIG", func, "readonly");
+  return CreateSubCommands(ctx, configCommand, subcommands, sizeof(subcommands) / sizeof(SubCommand));
+}
+
+static int RegisterAllDebugCommands(RedisModuleCtx* ctx, RedisModuleCommand *debugCommand) {
+  int rc = RegisterDebugCommands(debugCommand);
+  if (rc != REDISMODULE_OK) {
+    return rc;
+  }
+  return RegisterCoordDebugCommands(debugCommand);
+}
+
+static int RegisterCursorCommands(RedisModuleCtx* ctx, RedisModuleCommand *cursorCommand);
+
+#define RS_WRITE_FLAGS_DEFAULT(flags) IsEnterprise() ? flags " " PROXY_FILTERED : flags
 
-  return rc;
+static int CreateSearchCommands(RedisModuleCtx *ctx, const SearchCommand *commands, size_t count) {
+  for (size_t i = 0; i < count; i++) {
+    const SearchCommand *command = &commands[i];
+    if (CreateSearchCommand(ctx, command) != REDISMODULE_OK) {
+      RedisModule_Log(ctx, "warning", "Could not create search command %s", command->name);
+      return REDISMODULE_ERR;
+    }
+  }
+  return REDISMODULE_OK;
 }
 
+#define DEFINE_COMMAND_KEYS(firstkey_, lastkey_, keystep_) \
+  (CommandKeys){.firstkey = firstkey_, .lastkey = lastkey_, .keystep = keystep_}
+
+#define DEFINE_COMMAND(name_, func_, flags_, callback_, callback_type_, acl_, register_condition_, keys_) \
+  { .name = name_, .flags = flags_, .shouldRegister = register_condition_,                                \
+    .aclCategories = acl_, .handler = func_, .callback.ptr = callback_,                                   \
+    .selectedCallbackType = callback_type_,                                                               \
+    .position = keys_}
```

Comment:
let's have all the macros in one, dedicate place. Either in this file, or macro file